### DISCUSSION TITLE
Bug 🐛: `node/no-missing-imports` fails

### DISCUSSION THREAD

**Discussion by jasikpark** [avajs/ava#2866] [Q&A]
Please provide details about:

* What you're trying to do
* What happened
* What you expected to happen

Please share relevant sample code. Or better yet, provide a link to a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).

We'll also need your AVA configuration (in `package.json` or `ava.config.*` configuration files) and how you're invoking AVA. Share the installed AVA version (get it by running `npx ava --version`).

`yarn ava --version` returns `0.0.8` though I have `"ava": "^4.0.0-alpha.2",` listed in my devDependencies

I'm using ava for [prettier-plugin-astro](https://github.com/snowpackjs/prettier-plugin-astro), and I've run into a problem where [eslint:node/no-missing-imports](https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-missing-import.md) is failing for me, I believe because ava 4 does not define a `main` key? I'm not quite sure though. If I use `import test from 'ava/entrypoints/main.mjs';` the lint is happy though.

Is there something I can improve with my setup to fix this rule other than hardcoding the file import / disabling the rule for that line?

**Comment by novemberborn** [comment]
Could you try with the main branch, by installing directly from GitHub? There have been a lot of unreleased changes.

**Comment by hildjj** [comment]
This is still happening with ava 5.1.0

**Reply by novemberborn** [reply]
The most recent version of the plugin (https://github.com/mysticatea/eslint-plugin-node/blob/f45c6149be7235c0f7422d1179c25726afeecd83/docs/rules/no-missing-import.md) suggests it doesn't understand ESM or the module lookup that Node.js now supports. I think the plugin may simply be outdated.

**Comment by hildjj** [comment]
The issue here is https://github.com/browserify/resolve/issues/222

Ava does not provide a `main` entry in package.json, relying on `exports`.  The [resolve](https://github.com/browserify/resolve/) library that is used by [node/no-missing-import](https://github.com/mysticatea/eslint-plugin-node/blob/f45c6149be7235c0f7422d1179c25726afeecd83/docs/rules/no-missing-import.md) does not support this.  Ava is the first dependency I have which has this characteristic, which is why it's the only one having this issue.
